src/util/evalUnits.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+const si = [
+  { value: -1e24, symbol: 'y' },
+  { value: -1e21, symbol: 'z' },
+  { value: -1e18, symbol: 'a' },
+  { value: -1e15, symbol: 'f' },
+  { value: -1e12, symbol: 'p' },
+  { value: -1e9, symbol: 'n' },
+  { value: -1e6, symbol: 'μ' },
+  { value: -1e3, symbol: 'm' },
+  { value: 1, symbol: '' },
+  { value: 1e3, symbol: 'k' },
+  { value: 1e6, symbol: 'M' },
+  { value: 1e9, symbol: 'G' },
+  { value: 1e12, symbol: 'T' },
+  { value: 1e15, symbol: 'P' },
+  { value: 1e18, symbol: 'E' },
+  { value: 1e21, symbol: 'Z' },
+  { value: 1e24, symbol: 'Y' }
+];
+
+const floats = /^[0-9]*[.,]{1}[0-9]*$/;
+const ints = /^[0-9]+$/;
+const alphaFloats = /^[0-9]*[.,]{1}[0-9]*[a-zA-Z]{1}$/;
+const alphaInts = /^[0-9]*[a-zA-Z]{1}$/;
+
+export enum EvalMessages {
+  GIBBERISH = 'Input is not correct. Use numbers, floats or expression (e.g. 1k, 1.3m)',
+  SUCCESS = '',
+  SYMBOL_ERROR = 'Provided symbol is not correct',
+  GENERAL_ERROR = 'Check your input. Something went wrong'
+}
+
+/**
+ * A function that identifes integer/float(comma or dot)/expressions (such as 1k)
+ * and converts to actual value (or reports an error).
+ * @param {string} input
+ * @returns {[number | null, string]} an array of 2 items
+ * the first is the actual calculated number (or null if none) while
+ * the second is the message that should appear in case of error
+ */
+export function evalUnits(input: string): [number | null, string] {
+  if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
+    return [null, EvalMessages.GIBBERISH];
+  }
+
+  if (floats.test(input) || ints.test(input)) {
```

Comment:
I'm not sure if regex are the right solution for floats and ints. Couldn't we do;
```
const isFloat = parseFloat(value).toString() == value;
const isInt = parseInt(value).toString() == value;
```
?

Reply:
Not really.. both are doing very specific parsing that does not fit to our case. E.g. parseFloat will parse correctly 10.33 but if , decimal symbol is used then it will parse it to 10. Thus can lead to severe mistakes. The Regex combination above ensures that you have specific format for int, floats, string int, string floats and parses correctly the . and , decimal symbols. This way we keep textfield as input but reduce possible mistakes that user may do

Reply:
for integers you can rely on`Number.isInteger()` instead of a regex

Reply:
I was diving a little bit more and found this old repo that solved this with the following code:

```
    // A numeric is a variable that contains a numeric value, regardless its type
    // It can be a String containing a numeric value, exponential notation, or a Number object
    // See here for more discussion: http://stackoverflow.com/questions/18082/validate-numbers-in-javascript-isnumeric/1830844#1830844
    isNumeric: function(n) {
      return !isNaN(parseFloat(n)) && isFinite(n);
    },

    // An integer contains an optional minus sign to begin and only the digits 0-9
    // Objects that can be parsed that way are also considered ints, e.g. "123"
    // Floats that are mathematically equal to integers are considered integers, e.g. 1.0
    // See here for more discussion: http://stackoverflow.com/questions/1019515/javascript-test-for-an-integer
    isInteger: function(i) {
      return _.isNumeric(i) && i % 1 === 0;
    },

    // A float is a numbr that is not an integer.
    isFloat: function(n) {
      return _.isNumeric(n) && !_.isInteger(n);
    },
```

All the functions called from `isNumeric` are available in the  global `Numbers` object, so we don't have to re define this .

Reply:
> This can lead to severe mistakes. 

I agree and that's why I made the comment. You check that it's float by a regex, but then you use `parseFloat` on it, which might end up with different results.

Reply:
> but then you use parseFloat

This completely slipped. It is strange that tests did not get it (note: test was wrong so it was not capturing it - fixed it as well). Thank you - I used the replace of decimal symbol to make sure works in all cases